### PR TITLE
Add separate nsg for the router

### PR DESCRIPTION
**Pull request by kargakis**
Moved from https://github.com/kargakis/acs-engine/pull/67

Fixes https://github.com/Azure/acs-engine/issues/2684

@jackfrancis @pweil- @jim-minter

### CODE REVIEW

**Comment by jackfrancis** [comment]
Should we close PR #2604 in favor of this?

**Comment by kargakis** [comment]
> Should we close PR #2604 in favor of this?

Yes

File: parts/k8s/kubernetesmasterresources.t [hunk 1]
```diff
@@ -464,11 +483,17 @@
...
         },
         "dependsOn": [
+  {{if IsKubernetes}}
   {{if .MasterProfile.IsCustomVNET}}
           "[variables('nsgID')]"
   {{else}}
           "[variables('vnetID')]"
   {{end}}
+  {{else}}
+  {{if not .MasterProfile.IsCustomVNET}}
```
Comment (jim-minter) [inline parts/k8s/kubernetesmasterresources.t]: are you missing `"[variables('nsgID')]"` here?

**Comment by jim-minter** [comment]
I think there's one mistake, otherwise looks good

File: parts/k8s/kubernetesmasterresources.t (hunk 1 above)
Comment (kargakis) [reply parts/k8s/kubernetesmasterresources.t]: yes, added

File: parts/k8s/kubernetesagentresourcesvmas.t [hunk 2]
```diff
@@ -5,19 +5,40 @@
         "name": "loop"
       },
       "dependsOn": [
+{{if IsKubernetes}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesagentresourcesvmas.t]: Let's change this to `not OpenShift` to be consistent with #2765

File: parts/k8s/kubernetesagentresourcesvmas.t [hunk 3]
```diff
@@ -5,19 +5,40 @@
...
+      "[resourceId('Microsoft.Network/networkSecurityGroups', 'router-nsg')]"
+{{else}}
+      "[variables('nsgID')]"
+{{end}}
 {{end}}
       ],
       "location": "[variables('location')]",
       "name": "[concat(variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
       "properties": {
+{{if IsKubernetes}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesagentresourcesvmas.t]: ditto

File: parts/k8s/kubernetesmasterresources.t [hunk 4]
```diff
@@ -177,10 +216,13 @@
           {
             "name": "[variables('subnetName')]",
             "properties": {
-              "addressPrefix": "[variables('subnet')]",
+              "addressPrefix": "[variables('subnet')]"
+{{if IsKubernetes}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesmasterresources.t]: same here

File: parts/k8s/kubernetesmasterresources.t [hunk 5]
```diff
@@ -159,10 +197,11 @@
     {
       "apiVersion": "[variables('apiVersionDefault')]",
       "dependsOn": [
-        "[concat('Microsoft.Network/networkSecurityGroups/', variables('nsgName'))]"
 {{if not IsAzureCNI}}
-        ,
-        "[concat('Microsoft.Network/routeTables/', variables('routeTableName'))]"
+        "[concat('Microsoft.Network/routeTables/', variables('routeTableName'))]"{{if IsKubernetes}},{{end}}
+{{end}}
+{{if IsKubernetes}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesmasterresources.t]: and here

File: parts/k8s/kubernetesmasterresources.t [hunk 6]
```diff
@@ -464,11 +483,18 @@
           "name": "nicLoopNode"
         },
         "dependsOn": [
+  {{if IsKubernetes}}
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetesmasterresources.t]: and here

**Comment by kargakis** [comment]
@CecileRobertMichon updated

**Pull request merged**